src/python_testing/matter_testing_support.py

```diff
@@ -265,6 +265,15 @@ def wait_for_event_report(self, expected_event: ClusterObjects.ClusterEvent, tim
         asserts.assert_equal(res.Header.EventId, expected_event.event_id, "Expected event ID not found in event report")
         return res.Data
 
+    def wait_for_no_event_report(self, timeout: int = 10):
+        """This function allows a test script to block waiting for the specific event to arrive with a timeout.
+           It returns the event data so that the values can be checked."""
```

Comment:
It doesn't return anything? the comment looks copy and paste from above.

Also the docstring doesn't tell us that it will automatically fail if it does get an event report.
Maybe the function should be `wait_for_event_expect_no_report()`